### ISSUE TITLE
Why does React Native Map require a valid Google Maps API key to render custom tiles on Android?

### ISSUE THREAD

**Issue by junjiIshii** [react-native-maps/react-native-maps#5156]
### Summary

I have been using the React Native Map library, under the assumption that it functions similarly to libraries like Leaflet. I believed that the library provides functionality for interacting with maps and that it would allow me to display maps using custom tiles or by using a service like Google Maps API, depending on my needs. Specifically, I expected that by simply providing a tile server, I could manage costs effectively by only paying for the tile server usage.

However, I encountered an issue on Android where the map does not function correctly unless I include a valid Google Maps API key in AndroidManifest.xml. Despite using custom tiles that should be independent of Google Maps, the map fails to render without the key. I've also noticed that [previous issues](https://github.com/react-native-maps/react-native-maps/issues/773) suggest that requests are still being made to the Google Maps Platform.

### Reproducible sample code

```TSX
--- src/App.tsx
<MapView
style={{flex:1}}
mapType="none"
initialRegion={{
  latitude: 42,
  longitude: 30,
  latitudeDelta: 1,
  longitudeDelta: 1
}}
>
<UrlTile
  tileSize={512}
  urlTemplate="https://api.mapbox.com/styles/v1/fugafuga/hogehoge/tiles/256/{z}/{x}/{y}?access_token=xxxxx"
  maximumZ={19}
/> 
// I used map box but you can select other custom tiles, for example OSM, MapTiler and so on.
</MapView>
---

--- android/app/src/main/AndroidManifest.xml
<meta-data 
  android:name="com.google.android.geo.API_KEY"
  android:value="" />
// Anyway I added this tag because app can not build without this tag. I set value as blank because I will not use API KEY (google map platform access)
---
```

### Steps to reproduce

1. Create a new React Native application.
2. Set up React Native Map with a custom tile server (e.g., Mapbox or OpenStreetMap).
3. In AndroidManifest.xml, include the <meta-data android:name="com.google.android.geo.API_KEY" android:value=""/> tag, but leave the value empty.
4. If using an emulator, recreate the emulator and clear any cache. For physical devices, uninstall the app and rebuild from scratch to ensure no cached API keys are present.
5. Run the application on an Android device or emulator. Map component shows blank map but google logo locates under left.

### Expected result

I expected the library to handle map interaction independently, allowing for custom tiles to be used without requiring a Google Maps API key.

### Actual result

It requires valid google map api key. If I invalid api key, custom tile does not show. It shows blank gray map but google logo locates under left.

### React Native Maps Version

1.8.0

### What platforms are you seeing the problem on?

Android

### React Native Version

0.72.6

### What version of Expo are you using?

Not using Expo

### Device(s)

any android emulators and devices

### Additional information

Questions

1. Is the Google Maps SDK being used to render the map view, even when custom tiles are provided?
2. If so, is there any way to bypass this requirement when using custom tiles to avoid potential API charges?
3. If the API key is indeed necessary and charges could apply, should this be documented clearly to inform developers about the potential costs associated with using this library on Android?

Understanding these points is critical for developers who are trying to manage costs by using custom tile servers instead of relying on Google Maps.

**Comment by ssommelet21** [comment]
Hello junjiIshii

For your information, you can use Google Maps tiles, and it won't cost you anything. It's 100 % free, no limit.

Check the :

https://developers.google.com/maps/documentation/android-sdk/usage-and-billing

A Google map object, not loaded with a map ID, in a Maps SDK for Android or Maps SDK for iOS mobile application.

0.00 USD !!

This is not the 7$ for 1000 requests for Web API :

https://mapsplatform.google.com/pricing/

**Comment by RyanTG** [comment]
Seems like this should be moved to discussions. Not a bug.

**Comment by jinixx** [comment]
> Hello junjiIshii
> 
> For your information, you can use Google Maps tiles, and it won't cost you anything. It's 100 % free, no limit.
> 
> Check the :
> 
> https://developers.google.com/maps/documentation/android-sdk/usage-and-billing
> 
> A Google map object, not loaded with a map ID, in a Maps SDK for Android or Maps SDK for iOS mobile application.
> 
> 0.00 USD !!
> 
> This is not the 7$ for 1000 requests for Web API :
> 
> https://mapsplatform.google.com/pricing/

This does not answer the question, the requirement is to use some custom tiles from other providers. I have the same question, why a Google API key is needed? Can it be avoided?

**Comment by RyanTG** [comment]
No, that did answer the original question. The original question was about managing costs. As noted, including the google maps api key has no impact on costs.

I don’t think it can be avoided. It’s something that’s been discussed a lot here and in other forums.

**Comment by jinixx** [comment]
> No, that did answer the original question. The original question was about managing costs. As noted, including the google maps api key has no impact on costs.
> 
> I don’t think it can be avoided. It’s something that’s been discussed a lot here and in other forums.

I saw the other threads now. Looks like there is no way to avoid the Google API key.

**Comment by neibips** [comment]
Any updates in 2025?

**Comment by Michota** [comment]
bump

**Comment by RyanTG** [comment]
What are you bumping? This is answered.

**Comment by Michota** [comment]
I don't consider it answered really.

**Comment by RyanTG** [comment]
Again, the original concern is stated as:

> Specifically, I expected that by simply providing a tile server, I could manage costs effectively by only paying for the tile server usage.

And it was clarified that supplying a google maps api key does NOT incur any costs. It's unlimited free use.

Does this answer your question: This package supports two providers: apple maps and google maps. A provider is required, even when using a custom overlay.

**Comment by Michota** [comment]
Yes, it does. Thank you.

czw., 14 sie 2025, 19:08 użytkownik Ryan TG ***@***.***>
napisał:
